execution/engine.go

```diff
@@ -273,6 +274,19 @@ func (e *Engine) IsEligibleForProposerBonus(marketID string, value *num.Uint) bo
 	return value.ToDecimal().GreaterThan(quantum.Mul(e.npv.minLpStakeQuantumMultiple))
 }
 
+// MarketsExpectKey checks if the currently existing markets use the provided key in their subscriptions.
+func (e *Engine) MarketsExpectKey(key string) bool {
```

Comment:
Based on my previous comments, the algorithm is a bit more complex than this. 

We consider the OracleData to match an OracleSpec when all the public keys that signed the oracle data are defined in the oracle spec.

So getting a **list** of the _watched public keys_ won't be enough. You will need a more complex data structure.

For example:
* Market A listens to 0x11111
* Market B listens to 0x22222
* An oracle data signed by 0x11111 will match
* But an oracle data signed by 0x11111 and 0x22222 won't match, because no one is listening at **both keys at the same time**

Reply:
Yes, this has been addressed here https://github.com/vegaprotocol/vega/pull/5534#issuecomment-1177353456
Thanks!